homeassistant/components/renault/sensor.py

```diff
@@ -0,0 +1,277 @@
+"""Support for Renault sensors."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    DEVICE_CLASS_BATTERY,
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_TEMPERATURE,
+    LENGTH_KILOMETERS,
+    PERCENTAGE,
+    POWER_KILO_WATT,
+    TEMP_CELSIUS,
+    TIME_MINUTES,
+    VOLUME_LITERS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.icon import icon_for_battery_level
+from homeassistant.util import slugify
+
+from .const import (
+    DEVICE_CLASS_CHARGE_MODE,
+    DEVICE_CLASS_CHARGE_STATE,
+    DEVICE_CLASS_PLUG_STATE,
+    DOMAIN,
+)
+from .renault_entities import (
+    RenaultBatteryDataEntity,
+    RenaultChargeModeDataEntity,
+    RenaultCockpitDataEntity,
+    RenaultDataEntity,
+    RenaultHVACDataEntity,
+)
+from .renault_hub import RenaultHub
+from .renault_vehicle import RenaultVehicleProxy
+
+ATTR_BATTERY_AVAILABLE_ENERGY = "battery_available_energy"
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Renault entities from config entry."""
+    proxy: RenaultHub = hass.data[DOMAIN][config_entry.unique_id]
+    entities = await get_entities(proxy)
+    async_add_entities(entities)
+
+
+async def get_entities(proxy: RenaultHub) -> list[RenaultDataEntity]:
+    """Create Renault entities for all vehicles."""
+    entities = []
+    for vehicle in proxy.vehicles.values():
+        entities.extend(await get_vehicle_entities(vehicle))
+    return entities
+
+
+async def get_vehicle_entities(vehicle: RenaultVehicleProxy) -> list[RenaultDataEntity]:
```

Comment:
This doesn't need to be a coroutine function since we don't await inside.

Reply:
If I move up the list, I still end up with `async def async_setup_entry`
Is there a non-async version of this method?

Reply:
No, that function is part of the API, so we can't change that one.